unit-tests/src/test/java/org/eclipse/collections/impl/factory/SetsTest.java

```diff
@@ -762,6 +762,21 @@ public void cartesianProduct()
     }
 
     @Test
+    public void cartesianProductWithFunction()
+    {
+        MutableSet<Integer> set1 = UnifiedSet.newSetWith(1, 2);
```

Comment:
Regarding adding the API on Set interfaces, makes sense to first add it as a static utility.
Thanks!